enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestionResult.tsx

```diff
@@ -0,0 +1,128 @@
+import cx from "classnames";
+import { t } from "ttag";
+
+import type { SdkClickActionPluginsConfig } from "embedding-sdk";
+import { SdkError } from "embedding-sdk/components/private/PublicComponentWrapper";
+import { ResetButton } from "embedding-sdk/components/private/ResetButton";
+import { getDefaultVizHeight } from "embedding-sdk/lib/default-height";
+import { useSdkSelector } from "embedding-sdk/store";
+import { getPlugins } from "embedding-sdk/store/selectors";
+import CS from "metabase/css/core/index.css";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import {
+  navigateToNewCardInsideQB,
+  updateQuestion,
+} from "metabase/query_builder/actions";
+import QueryVisualization from "metabase/query_builder/components/QueryVisualization";
+import { FilterHeader } from "metabase/query_builder/components/view/ViewHeader/components";
+import {
+  getCard,
+  getFirstQueryResult,
+  getQueryResults,
+  getQuestion,
+  getUiControls,
+} from "metabase/query_builder/selectors";
+import { Box, Flex, Group, Loader, Stack } from "metabase/ui";
+import { getEmbeddingMode } from "metabase/visualizations/click-actions/lib/modes";
+
+const returnNull = () => null;
+
+interface InteractiveQuestionResultProps {
+  isQuestionLoading: boolean;
+  onNavigateBack: () => void;
+  withResetButton?: boolean;
+  onResetButtonClick: () => void;
+  withTitle?: boolean;
+  customTitle?: React.ReactNode;
+
+  height?: string | number;
+  componentPlugins?: SdkClickActionPluginsConfig;
+}
+
+export const InteractiveQuestionResult = ({
+  isQuestionLoading,
+  componentPlugins,
+  onNavigateBack,
+  height,
+  withResetButton,
+  onResetButtonClick,
+  withTitle,
+  customTitle,
+}: InteractiveQuestionResultProps): React.ReactElement => {
+  const dispatch = useDispatch();
+
+  const globalPlugins = useSdkSelector(getPlugins);
+  const question = useSelector(getQuestion);
+  const card = useSelector(getCard);
+  const result = useSelector(getFirstQueryResult);
+  const uiControls = useSelector(getUiControls);
+  const queryResults = useSelector(getQueryResults);
+
+  const { isRunning: isQueryRunning } = uiControls;
+
+  if (isQuestionLoading || isQueryRunning || !queryResults) {
+    return <Loader data-testid="loading-spinner" />;
```

Comment:
Should we use the SDK's Loader component here (overrideable by SDK users), or does that not apply here?